src/main/java/com/google/cloud/spanner/pgadapter/statements/TruncateStatement.java

```diff
@@ -0,0 +1,155 @@
+  static final class ParsedTruncateStatement {
+    static final class Builder {
+      final ImmutableList.Builder<TableOrIndexName> tables = ImmutableList.builder();
+      boolean only;
+      boolean star;
+      boolean restartIdentity;
+      boolean cascade;
+    }
+
+    final ImmutableList<TableOrIndexName> tables;
+    final boolean only;
+    final boolean star;
+    final boolean restartIdentity;
+    final boolean cascade;
+
+    private ParsedTruncateStatement(Builder builder) {
+      this.tables = builder.tables.build();
+      this.only = builder.only;
+      this.star = builder.star;
+      this.restartIdentity = builder.restartIdentity;
+      this.cascade = builder.cascade;
+    }
+  }
+
+  private final ParsedTruncateStatement truncateStatement;
+
+  public TruncateStatement(
+      ConnectionHandler connectionHandler,
+      OptionsMetadata options,
+      String name,
+      ParsedStatement parsedStatement,
+      Statement originalStatement) {
+    super(
+        name,
+        new IntermediatePreparedStatement(
+            connectionHandler,
+            options,
+            name,
+            NO_PARAMETER_TYPES,
+            parsedStatement,
+            originalStatement),
+        NO_PARAMS,
+        ImmutableList.of(),
+        ImmutableList.of());
+    this.truncateStatement = parse(originalStatement.getSql());
+  }
+
+  @Override
+  public String getCommandTag() {
+    return "TRUNCATE TABLE";
+  }
+
+  @Override
+  public StatementType getStatementType() {
+    return StatementType.DDL;
+  }
+
+  @Override
+  public void executeAsync(BackendConnection backendConnection) {
+    this.executed = true;
+    setFutureStatementResult(backendConnection.execute(this));
+  }
+
+  public List<Statement> getDeleteStatements() {
+    List<Statement> statements = new ArrayList<>(truncateStatement.tables.size());
+    for (TableOrIndexName table : truncateStatement.tables) {
+      statements.add(Statement.of("delete from " + table.toString()));
+    }
+    return statements;
+  }
+
+  @Override
+  public Future<StatementResult> describeAsync(BackendConnection backendConnection) {
+    // Return null to indicate that this TRUNCATE statement does not return any
+    // RowDescriptionResponse.
+    return Futures.immediateFuture(null);
+  }
+
+  @Override
+  public IntermediatePortalStatement createPortal(
+      String name,
+      byte[][] parameters,
+      List<Short> parameterFormatCodes,
+      List<Short> resultFormatCodes) {
+    // TRUNCATE does not support binding any parameters, so we just return the same statement.
+    return this;
+  }
+
+  static ParsedTruncateStatement parse(String sql) {
+    Preconditions.checkNotNull(sql);
+
+    SimpleParser parser = new SimpleParser(sql);
+    ParsedTruncateStatement.Builder builder = new Builder();
+    if (!parser.eatKeyword("truncate")) {
+      throw PGExceptionFactory.newPGException("not a valid TRUNCATE statement: " + sql);
+    }
+    parser.eatKeyword("table");
```

Comment:
Should we make this a static string to reduce memory?

Reply:
That is actually not necessary in Java. String literals are automatically 'interned' (added to the internal String pool of the JVM), which means that repeating the same string literal throughout an application will not use more memory than creating a single String constant and reuse that. See https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#intern--